grassroots-frontend/src/main.tsx

```diff
@@ -15,6 +15,7 @@ declare module "@tanstack/react-router" {
   }
 }
 
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 createRoot(document.getElementById("root")!).render(
```

Comment:
I've found this to be a useful error to have. It adds a little friction when using the `!`. Enough to stop its unnecessary use, but not enough to stop legitimate uses like the above.

Reply:
The other approach I've used (before I knew ! existed) is to define:

```javascript
export function el(id: string, message: string | undefined): Element {
  const result = document.getElementById(id);
  if (result) {
    return result;
  }
  throw new Error(message ?? `Couldn't find element with id ${id}`);
}
```

I don't find myself using ! in many other places. I'm happy either way.